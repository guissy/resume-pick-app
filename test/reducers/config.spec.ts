import { AnyAction } from 'redux';
import counterReducer, {
  initConfigAsync,
  resetConfigAsync,
} from '../../app/features/configSlice';

describe('reducers', () => {
  describe('config', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle initConfigAsync', () => {
      expect(
        counterReducer({ configs: { default: [] } }, { type: initConfigAsync })
      ).toMatchSnapshot();
    });

    it('should handle resetConfigAsync', () => {
      expect(
        counterReducer({ configs: { default: [] } }, { type: resetConfigAsync })
      ).toMatchSnapshot();
    });
  });
});
