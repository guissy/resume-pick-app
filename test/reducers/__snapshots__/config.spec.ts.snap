// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle initConfigAsync 1`] = `
Object {
  "configs": Object {
    "default": Array [],
  },
}
`;

exports[`reducers counter should handle initial state 1`] = `
Object {
  "configs": Object {},
}
`;

exports[`reducers counter should handle resetConfigAsync 1`] = `
Object {
  "configs": Object {
    "default": Array [],
  },
}
`;
